go/libraries/doltcore/sqle/enginetest/dolt_engine_test.go

```diff
@@ -2760,45 +2760,56 @@ func TestDoltStorageFormatPrepared(t *testing.T) {
 
 func TestThreeWayMergeWithSchemaChangeScripts(t *testing.T) {
 	skipOldFormat(t)
-	t.Run("right to left merges", func(t *testing.T) {
-		for _, script := range ThreeWayMergeWithSchemaChangeTestScripts {
-			func() {
-				h := newDoltHarness(t)
-				defer h.Close()
-				enginetest.TestScript(t, h, convertMergeScriptTest(script, false))
-			}()
-		}
-	})
-	t.Run("left to right merges", func(t *testing.T) {
-		for _, script := range ThreeWayMergeWithSchemaChangeTestScripts {
-			func() {
-				h := newDoltHarness(t)
-				defer h.Close()
-				enginetest.TestScript(t, h, convertMergeScriptTest(script, true))
-			}()
-		}
-	})
+	runMergeScriptTestsInBothDirections(t, SchemaChangeTestsBasicCases, "basic cases", false)
+	runMergeScriptTestsInBothDirections(t, SchemaChangeTestsForDataConflicts, "data conflicts", false)
+	runMergeScriptTestsInBothDirections(t, SchemaChangeTestsCollations, "collation changes", false)
+	runMergeScriptTestsInBothDirections(t, SchemaChangeTestsConstraints, "constraint changes", false)
+	runMergeScriptTestsInBothDirections(t, SchemaChangeTestsTypeChanges, "type changes", false)
+	runMergeScriptTestsInBothDirections(t, SchemaChangeTestsSchemaConflicts, "schema conflicts", false)
 }
 
 func TestThreeWayMergeWithSchemaChangeScriptsPrepared(t *testing.T) {
 	skipOldFormat(t)
-	t.Run("right to left merges", func(t *testing.T) {
-		for _, script := range ThreeWayMergeWithSchemaChangeTestScripts {
-			func() {
-				h := newDoltHarness(t)
-				defer h.Close()
-				enginetest.TestScript(t, h, convertMergeScriptTest(script, false))
-			}()
-		}
-	})
-	t.Run("left to right merges", func(t *testing.T) {
-		for _, script := range ThreeWayMergeWithSchemaChangeTestScripts {
-			func() {
-				h := newDoltHarness(t)
-				defer h.Close()
-				enginetest.TestScript(t, h, convertMergeScriptTest(script, true))
-			}()
-		}
+	runMergeScriptTestsInBothDirections(t, SchemaChangeTestsBasicCases, "basic cases", true)
+	runMergeScriptTestsInBothDirections(t, SchemaChangeTestsForDataConflicts, "data conflicts", true)
+	runMergeScriptTestsInBothDirections(t, SchemaChangeTestsCollations, "collation changes", true)
+	runMergeScriptTestsInBothDirections(t, SchemaChangeTestsConstraints, "constraint changes", true)
+	runMergeScriptTestsInBothDirections(t, SchemaChangeTestsTypeChanges, "type changes", true)
+	runMergeScriptTestsInBothDirections(t, SchemaChangeTestsSchemaConflicts, "schema conflicts", true)
+}
+
+// runMergeScriptTestsInBothDirections creates a new test run, named |name|, and runs the specified merge |tests|
+// in both directions (right to left merge, and left to right merge). If
+// |runAsPrepared| is true then the test scripts will be run using the prepared
+// statement test code.
+func runMergeScriptTestsInBothDirections(t *testing.T, tests []MergeScriptTest, name string, runAsPrepared bool) {
+	t.Run(name, func(t *testing.T) {
+		t.Run("right to left merges", func(t *testing.T) {
+			for _, script := range tests {
+				func() {
```

Comment:
does this do anything? I think the callback into the test handler is already parallelized

Reply:
the nested `t.Run`s just give a little more structure to the test results; that inner `func()` on line 2789 is really just there to scope the call to `defer` a couple lines down. It's not the most obvious pattern though, so I'll add a comment. Let me know if you meant something else.